kuma/users/views.py

```diff
@@ -599,31 +599,31 @@ def get_form(self, form_class=None):
         form = super(SignupView, self).get_form(form_class)
         form.fields["email"].label = _("Email address")
 
-        User = get_user_model()
-
-        # When no username is provided, default to the local-part of the email address
-        if form.initial.get("username", "") == "":
-            email = form.initial.get("email", "")
-            if isinstance(email, tuple):
-                email = email[0]
-            suggested_username = suggested_username_base = email.split("@")[0]
+        # We should only see GitHub/Google users.
+        assert self.sociallogin.account.provider in ("github", "google")
+
+        initial_username = form.initial.get("username")
+
+        # When no username is provided, try to derive one from the email address.
+        if not initial_username:
+            email = form.initial.get("email")
+            if email:
+                if isinstance(email, tuple):
+                    email = email[0]
+                initial_username = email.split("@")[0]
+
+        if initial_username:
+            # Find a new username if it clashes with an existing username.
             increment = 1
-            while User.objects.filter(username__iexact=suggested_username).exists():
+            User = get_user_model()
+            initial_username_base = initial_username
+            while User.objects.filter(username__iexact=initial_username).exists():
                 increment += 1
-                suggested_username = f"{suggested_username_base}{increment}"
-            form.initial["username"] = suggested_username
-
-        self.matching_user = None
-        initial_username = form.initial.get("username", None)
+                initial_username = f"{initial_username_base}{increment}"
+        else:
+            initial_username = ""
 
-        # For GitHub/Google users, see if we can find matching user by username
-        assert self.sociallogin.account.provider in ("github", "google")
-        try:
-            self.matching_user = User.objects.get(username=initial_username)
-            # deleting the initial username because we found a matching user
-            del form.initial["username"]
```

Comment:
With this, you can never get a KeyError on `form.initial['username']` any more. 👍🏼